s3fs/core.py

```diff
@@ -978,7 +978,8 @@ async def _exists(self, path):
                 if self._ls_from_cache(path):
                     return True
             except FileNotFoundError:
-                return False
```

Comment:
So we are getting this exception if we've succeded to list path parent, but haven't found matches. In that sense returning `False` makes sense here. But for versioned path we'll fail to match versioned name and will also get FileNotFoundError. Let me take another look...